src/autoupdate.c

```diff
@@ -112,7 +112,18 @@ int autoupdate_main(int argc, char **argv)
 		}
 		return (rc);
 	} else {
-		int rc = system("/usr/bin/systemctl is-enabled swupd-update.service > /dev/null");
+		/* In a container, "/usr/bin/systemctl" will return 1 with
+		 * "Failed to connect to bus: No such file or directory"
+		 * However /usr/bin/systemctl is-enabled ... will not fail, even when it
+		 * should. Check that systemctl is working before reporting the output
+		 * of is-enabled. */
+		int rc = system("/usr/bin/systemctl > /dev/null");
+		if (rc == 1) {
```

Comment:
For this case, I think you need to use `WEXITSTATUS()` to get the real return value after verifying `WIFEXITED()` returns non-zero.